src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -716,11 +738,13 @@ private static List<TestInputViolation> getExpectedViolationsForSpecificLineAbov
 
     private static List<TestInputViolation> getExpectedViolations(
                                               List<String> lines, int lineNo,
-                                              Matcher matcher) {
-        final int linesAbove =
+                                              Matcher matcher, boolean isAbove) {
+        final int violationLine =
             Integer.parseInt(matcher.group(2));
-        final int violationLineNum = lineNo - linesAbove + 1;
-        return getExpectedViolationsForSpecificLineAbove(lines,
+        final int violationLineNum =
+                isAbove ? lineNo - violationLine + 1
+                        : lineNo + violationLine + 1;
+        return getExpectedViolationsForSpecificLine(lines,
             lineNo, violationLineNum, matcher);
     }
```

Comment:
Had to modify this method to identify which violation message we're checking and also it'd be helpful if you can suggest a better name for: `violationLine`

I could also create a new method similar to the existing method, that way I would not need to modify the existing method but these methods would be very identical to each other, the only difference would be:

For checking "lines above message": `lineNo - linesAbove + 1`
For checking "lines below message": `lineNo - linesBelow + 1`